Scripts/Runtime/Debug/UnityTraceListener.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using UnityEngine;
+
+namespace Anvil.Unity.Debugging
+{
+    /// <summary>
+    /// A custom <see cref="System.Diagnostics.TraceListener"> for Unity, to allow the use of <see cref="System.Diagnostics"> functions.
+    /// By default, all of the .NET Debug/Trace functions have no effect in Unity, without a listener manually forwarding calls.
+    /// 
+    /// The most common and important functions enabled by this listener are:
+    /// - <see cref="System.Diagnostics.Debug.WriteLine"> (forwarded to UnityEngine.Debug.Log)
+    /// - <see cref="System.Diagnostics.Debug.Assert"> (forwarded to a thrown exception)
+    /// 
+    /// Note: Both <see cref="System.Diagnostics.Debug"> and <see cref="System.Diagnostics.Trace"> functions use the same
+    /// listener, the difference is that Debug functions become no-ops in release builds, while Trace always works.
+    /// </summary>
+    public class UnityTraceListener : System.Diagnostics.TraceListener
+    {
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.SubsystemRegistration)]
+        private static void Init()
+        {
+            System.Diagnostics.Trace.Listeners.Add(new UnityTraceListener());
```

Comment:
Do we get duplicates in here with Domain Reloading turned off and multiple compiles of the code? 

See: https://docs.unity3d.com/2021.3/Documentation/Manual/DomainReloading.html

My gut would say that this looks like we're adding a new instance to a static list each time which will lead to duplicates but I'm not sure how the Listeners work under the hood.

Reply:
Hmm I'm not sure, I'll test it out and add a static instance if necessary.

Reply:
I'm getting confusing results - with or without "Enter Play Mode Settings > Reload Domain" enabled in editor settings, this static function is only ever called once for me, unless I'm not triggering domain reloading correctly.

I.e. if I throw a log in that static `Init()` and start play mode, I see it log once - then if I make a code change and go back to Unity, it reloads scripts but I don't see that log again, and there's no duplicate listener.

I considered adding a `static s_Instance` anyway, but based on the above link, domain reloading would reset static fields anyway? I'm unsure how to handle this case - there doesn't seem to be anything going wrong, but I might be missing a use case. I will continue researching, but I'm open to suggestions.

Reply:
Ahhh my confusion comes from the editor settings - I thought Reload Domain was disabled by default because it's unchecked, but if "Enter Play Mode Options" is also unchecked, it defaults to enabled, so I wasn't actually testing the disabled state.

Would this be sufficient?
```cs
if (s_Instance == null)
{
    s_Instance = new UnityTraceListener();
    System.Diagnostics.Trace.Listeners.Add(s_Instance);
}
```